src/main/java/nl/moreniekmeijer/backendspringboottechiteasycontroller/controllers/TelevisionsController.java

```diff
@@ -1,11 +1,74 @@
 package nl.moreniekmeijer.backendspringboottechiteasycontroller.controllers;
 
+import nl.moreniekmeijer.backendspringboottechiteasycontroller.exceptions.IndexOutOfBoundsException;
+import nl.moreniekmeijer.backendspringboottechiteasycontroller.exceptions.NameTooLongException;
+import nl.moreniekmeijer.backendspringboottechiteasycontroller.exceptions.RecordNotFoundException;
+import nl.moreniekmeijer.backendspringboottechiteasycontroller.models.Television;
+import nl.moreniekmeijer.backendspringboottechiteasycontroller.repositories.TelevisionRepository;
+import org.springframework.data.domain.Sort;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
+@RestController
+@RequestMapping("/televisions")
 public class TelevisionsController {
 
-}
\ No newline at end of file
+    private final TelevisionRepository televisionRepository;
+
+    public TelevisionsController(TelevisionRepository televisionRepository) {
+        this.televisionRepository = televisionRepository;
+    }
+
+    @PostMapping
+    public ResponseEntity<Television> addTelevision(@RequestBody Television television) {
+        if (television.getName().length() > 20) {
+            throw new NameTooLongException(20);
+        } else {
+        televisionRepository.save(television);
+        }
+        return ResponseEntity.created(null).body(television);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<Television>> getTelevisions() {
+        List<Television> foundTelevisions = televisionRepository.findAll(Sort.by(Sort.Direction.ASC, "id"));
+        return ResponseEntity.ok(foundTelevisions);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Television> getTelevisionById(@PathVariable Long id) {
+        if (id < 0) {
```

Comment:
Deze is inprincipe niet nodig, omdat de optional leeg zou zijn als het negatief is :)